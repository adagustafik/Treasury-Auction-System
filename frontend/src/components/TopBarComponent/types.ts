export interface TopBarProp {
  children?: React.ReactNode;
}
