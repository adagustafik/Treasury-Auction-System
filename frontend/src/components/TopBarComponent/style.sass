@use '../../App'

.top-bar
    display: flex
    justify-content: space-between
    align-items: center
    height: 11.5vh
    background-color: App.$blackOverlay
    font-size: 2.1vh
    color: white

.top-bar-child
    padding: 2vh

.top-bar-child p
    margin: -0.3vh 

.top-bar a
    color: white

.top-bar a:hover
    color: App.$buttonColor   