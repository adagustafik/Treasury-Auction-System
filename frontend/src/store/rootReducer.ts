import { combineReducers } from "@reduxjs/toolkit";
import { auctionsReducer } from "./slices/auctions";
import { userReducer } from "./slices/user";
import { bidsReducer } from "./slices/bids";
import { purchasesReducer } from "./slices/purchases";

export const rootReducer = combineReducers({
  userReducer,
  auctionsReducer,
  purchasesReducer,
  bidsReducer,
});
