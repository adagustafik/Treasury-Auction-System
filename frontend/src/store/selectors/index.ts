export * from "./selectUser";
export * from "./selectAuctions";
export * from "./selectBids";
export * from "./selectPurchases";
