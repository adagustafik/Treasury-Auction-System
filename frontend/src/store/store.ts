import {
  configureStore,
  ThunkAction,
  Action,
  AnyAction,
  ThunkDispatch,
} from "@reduxjs/toolkit";
import thunkMiddleware from "redux-thunk";
import { rootReducer } from "./rootReducer";

export const store = configureStore({
  reducer: rootReducer,
  middleware: [thunkMiddleware],
});

export type AppDispatch = typeof store.dispatch &
  ThunkDispatch<RootState, null, AnyAction>;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
