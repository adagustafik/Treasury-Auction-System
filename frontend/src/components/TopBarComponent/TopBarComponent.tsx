import React from "react";
import Container from "react-bootstrap/Container";
import { Link } from "react-router-dom";
import { TopBarProp } from "./types";
import "./style.sass";
import { useAppSelector } from "../../store/hooks";
import { selectUser } from "../../store/selectors";

export function TopBarComponent({ children }: TopBarProp) {
  const { user } = useAppSelector(selectUser);

  return (
    <Container fluid className="top-bar">
      <Link to="/dashboard">
        <div className="top-bar-child">
          <p>TREASURY</p>
          <p>AUCTION</p>
          <p>SYSTEM</p>
        </div>
      </Link>
      {children && <div className="top-bar-child">{children}</div>}
      {!children && <div className="top-bar-child">{user}</div>}
    </Container>
  );
}
