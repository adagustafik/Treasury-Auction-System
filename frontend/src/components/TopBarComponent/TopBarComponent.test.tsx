import { Provider } from "react-redux";
import { BrowserRouter, Link } from "react-router-dom";
import renderer from "react-test-renderer";
import { store } from "../../store/store";
import { TopBarComponent } from "./TopBarComponent";

it("renders correctly without props", () => {
  const registerForm = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <TopBarComponent />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(registerForm).toMatchSnapshot();
});

it("renders correctly with Link node to login", () => {
  const registerFormLogin = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <TopBarComponent>
            <Link to="/login">LOGIN</Link>
          </TopBarComponent>
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(registerFormLogin).toMatchSnapshot();
});
