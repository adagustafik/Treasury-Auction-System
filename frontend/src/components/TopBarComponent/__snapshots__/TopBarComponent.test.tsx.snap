// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with Link node to login 1`] = `
<div
  className="top-bar container-fluid"
>
  <a
    href="/dashboard"
    onClick={[Function]}
  >
    <div
      className="top-bar-child"
    >
      <p>
        TREASURY
      </p>
      <p>
        AUCTION
      </p>
      <p>
        SYSTEM
      </p>
    </div>
  </a>
  <div
    className="top-bar-child"
  >
    <a
      href="/login"
      onClick={[Function]}
    >
      LOGIN
    </a>
  </div>
</div>
`;

exports[`renders correctly without props 1`] = `
<div
  className="top-bar container-fluid"
>
  <a
    href="/dashboard"
    onClick={[Function]}
  >
    <div
      className="top-bar-child"
    >
      <p>
        TREASURY
      </p>
      <p>
        AUCTION
      </p>
      <p>
        SYSTEM
      </p>
    </div>
  </a>
  <div
    className="top-bar-child"
  />
</div>
`;
